lab_2_tokenize_by_bpe/main.py

```diff
@@ -47,6 +97,32 @@ def merge_tokens(
     :param pair: a pair of tokens to be merged
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not(
+        isinstance(word_frequencies, dict)
+        and isinstance(pair, tuple)
+    ):
+        return None
+
+    merged_dict = {}
+
+    for word_tokens in word_frequencies:
+        word_tokens_list = list(word_tokens)
+        if pair[1] in word_tokens:
+            pair_token_index = []
+            for index in range(len(word_tokens) - 1):
+                token_pair = (word_tokens[index], word_tokens[index + 1])
+                if token_pair == pair:
+                    pair_token_index.append(index)
+
+            for pair_index in reversed(pair_token_index):
+                word_tokens_list[pair_index: pair_index + 2] = [''.join(pair)]
+
+            merged_dict[tuple(word_tokens_list)] = word_frequencies[word_tokens]
+
+        else:
+            merged_dict[word_tokens] = word_frequencies[word_tokens]
```

Comment:
does the merged dict contain all takens from word_frequencies?

Reply:
looks like now it contains only word_tokens that does not contain pair and new tokens

Reply:
In documentation of this function this is the way function works. It merges two tokens into one and doesn't leave both of them separately.